src/Sql/Vault/dbo/Stored Procedures/Cipher/Cipher_Delete_V2.sql

```diff
@@ -0,0 +1,73 @@
+﻿CREATE PROCEDURE [dbo].[Cipher_Delete_V2]
+    @Ids AS [dbo].[GuidIdArray] READONLY,
+    @UserId AS UNIQUEIDENTIFIER
+AS
+BEGIN
+    SET NOCOUNT ON
+
+    CREATE TABLE #Temp
+    (
+        [Id] UNIQUEIDENTIFIER NOT NULL,
+        [UserId] UNIQUEIDENTIFIER NULL,
+        [OrganizationId] UNIQUEIDENTIFIER NULL,
+        [Attachments] BIT NOT NULL
+    )
+
+    INSERT INTO #Temp
+    SELECT
+        [Id],
+        [UserId],
+        [OrganizationId],
+        CASE WHEN [Attachments] IS NULL THEN 0 ELSE 1 END
+    FROM
+        [dbo].[UserCipherDetails_V2](@UserId)
+    WHERE
+        [Edit] = 1
+        AND [Id] IN (SELECT * FROM @Ids)
+
+    -- Delete ciphers
+    DELETE
+    FROM
+        [dbo].[Cipher]
+    WHERE
+        [Id] IN (SELECT [Id] FROM #Temp)
+
+    -- Cleanup orgs
```

Comment:
Also while loops tend to be less resource intense and faster than cursors:

DECLARE @OrgId UNIQUEIDENTIFIER
DECLARE @OrgIndex INT = 1

-- Create a table variable to hold the distinct OrganizationId values
DECLARE @OrgTable TABLE (IndexId INT IDENTITY(1,1), OrganizationId UNIQUEIDENTIFIER)
INSERT INTO @OrgTable (OrganizationId)
SELECT DISTINCT [OrganizationId]
FROM #Temp
WHERE [OrganizationId] IS NOT NULL

WHILE EXISTS (SELECT 1 FROM @OrgTable WHERE IndexId = @OrgIndex)
BEGIN
    SELECT @OrgId = OrganizationId FROM @OrgTable WHERE IndexId = @OrgIndex
    EXEC [dbo].[Organization_UpdateStorage] @OrgId
    EXEC [dbo].[User_BumpAccountRevisionDateByOrganizationId] @OrgId
    SET @OrgIndex = @OrgIndex + 1
END